pagopa/io-app
# feat: [IOCOM-689] Move Third Part Data retrieval before Message Details screen

## Short description
This PR moves the logic to retrieve a message in a dedicated saga, while showing the Message Router Screen

## List of changes proposed in this pull request
- added a new reducer to handle the status of message data loading/aggregation
- added a new saga that implements the old Message Router Screen data retrieval logic. It also takes care of downloading the third party data if needed (and it marks the message as read only if such third party data have been successfully retrieved - same logic for SEND messages will come later in another PR)
- Message Router Screen: refactored in order to be a functional component and to just render itself, taking care only of the navigation and of the initial action dispatch. It also shows an error if the saga fails
- Fixed a condition where the user could get to a message details from a push notification without having accepted the message's preconditions
- MessageDetailsScreen details component does not retrieve Third Party Data on its own anymore (since the saga takes care of that now)

## How to test
Using the io-dev-api-server, check that any kind of message can be properly retrieved (standard messages, messages with payments, third party data messages, EU Covid messages and SEND messages)
